pennylane/transforms/metric_tensor.py

```diff
@@ -382,20 +387,36 @@ def qnode_execution_wrapper(self, qnode, targs, tkwargs):
     tkwargs.setdefault("device_wires", qnode.device.wires)
     mt_fn = self.default_qnode_wrapper(qnode, targs, tkwargs)
 
-    def _expand_fn(tape):
-        # Currently does not support it
-        tkwargs.pop("argnums", None)
-        return self.expand_fn(tape, *targs, **tkwargs)
+    def wrapper(*args, **kwargs):  # pylint: disable=too-many-branches
+        argnum = tkwargs.get("argnum", None)
+        argnums = tkwargs.get("argnums", None)
 
-    def wrapper(*args, **kwargs):
-        old_interface = qnode.interface
+        interface = qml.math.get_interface(*args)
+        trainable_params = qml.math.get_trainable_indices(args)
 
-        if old_interface == "auto":
-            qnode.interface = qml.math.get_interface(*args, *list(kwargs.values()))
+        if interface == "jax" and argnum is not None:
+            warnings.warn(
+                "argnum is deprecated with the Jax interface. You should use argnums instead."
+            )
+            tkwargs.pop("argnum")
+            argnums = argnum
 
-        cjac_fn = qml.transforms.classical_jacobian(qnode, expand_fn=_expand_fn)
+        argnums_ = None
 
-        if not qml.math.get_trainable_indices(args):
+        if interface == "jax" and not trainable_params:
+            if argnums is None:
+                argnums_ = [0]
+
+            else:
+                argnums_ = [argnums] if isinstance(argnums, int) else argnums
+
+            params = qml.math.jax_argnums_to_tape_trainable(
+                qnode, argnums_, self.expand_fn, args, kwargs
+            )
+            argnums_ = qml.math.get_trainable_indices(params)
+            kwargs["argnums"] = argnums_
+
+        if not trainable_params and argnums is None and argnums_ is None:

```

Comment:
If the above `if` statement is true, we will never have `argnums_ is None`, so we could do 
```suggestion
        elif not trainable_params and argnums is None and argnums_ is None:

```